src/test/java/it/gov/pagopa/initiative/statistics/events/consumers/CommandsMessagesListenerTest.java

```diff
@@ -0,0 +1,190 @@
+    @Autowired
+    private MerchantInitiativeCountersRepository merchantInitiativeCountersRepository;
+
+    @Test
+    void test() {
+        int validMessages = 100;
+        int notValidMessages = errorUseCasesNotify.size();
+        long maxWaitingMs = 30000;
+
+        List<String> commandsPayloads = new ArrayList<>(notValidMessages+validMessages);
+        commandsPayloads.addAll(IntStream.range(0,notValidMessages).mapToObj(i -> errorUseCasesNotify.get(i).getFirst().get()).toList());
+        commandsPayloads.addAll(buildValidPayloads(notValidMessages, notValidMessages+validMessages));
+
+        long timeStart=System.currentTimeMillis();
+        commandsPayloads.forEach(cp -> kafkaTestUtilitiesService.publishIntoEmbeddedKafka(topicCommands, null, null, cp));
+        long timePublishingEnd = System.currentTimeMillis();
+
+        waitForLastStorageChange(validMessages/2);
+        long timeEnd=System.currentTimeMillis();
+
+        System.out.printf("""
+                        ************************
+                        Time spent to send %d (%d + %d) messages (from start): %d millis
+                        Time spent to assert db stored count (from previous check): %d millis
+                        ************************
+                        Test Completed in %d millis
+                        ************************
+                        """,
+                commandsPayloads.size(),
+                validMessages,
+                notValidMessages,
+                timePublishingEnd - timeStart,
+                timeEnd - timePublishingEnd,
+                timeEnd - timeStart
+        );
+
+        checkRepositories();
+        checkErrorsPublished(notValidMessages, maxWaitingMs, errorUseCasesNotify);
+
+        System.out.printf("""
+                        ************************
+                        Time spent to send %d (%d + %d) messages (from start): %d millis
+                        Time spent to assert db stored count (from previous check): %d millis
+                        ************************
+                        Test Completed in %d millis
+                        ************************
+                        """,
+                commandsPayloads.size(),
+                validMessages,
+                notValidMessages,
+                timePublishingEnd - timeStart,
+                timeEnd - timePublishingEnd,
+                timeEnd - timeStart
+        );
+    }
+
+    private long waitForLastStorageChange(int n) {
+        long[] countSaved={0};
+        //noinspection ConstantConditions
+        TestUtils.waitFor(()->(countSaved[0]=merchantInitiativeCountersRepository.count()) == n, ()->"Expected %d saved merchant counters in db, read %d, DB elements %s".formatted(n, countSaved[0], merchantInitiativeCountersRepository.findAll().toString()), 60, 2000);
+        return countSaved[0];
+    }
+
+    private List<String> buildValidPayloads(int startValue, int messagesNumber) {
```

Comment:
messagesNumber is ambigous as parameter name used this way: you are using it as the end range, not as the number of elements created.

Rename it or change its usage